app/models/product.rb

```diff
@@ -1,4 +1,15 @@
 class Product < ApplicationRecord
   belongs_to :user
   validates_presence_of :name, :quantity, :expiry_date, :user_id
+
+  def list_expiring_products
+    User.all.each do |user|
+      products = list_expired_products(user.id, 7)
+    end
+  end
+
+  def list_expired_products(user_id, days)
```

Comment:
A note for the future: if you see yourself adding quite a lot of logic into a model than it is not _that_ directly related to the model itself, consider using [service objects](https://multithreaded.stitchfix.com/blog/2015/06/02/anatomy-of-service-objects-in-rails/) (that's one of my favourite articles).